artifactory_test.go

```diff
@@ -1866,6 +1866,30 @@ func TestSymlinkWildcardPathHandling(t *testing.T) {
 	cleanArtifactoryTest()
 }
 
+func TestUploadWithArchiveAndSymlink(t *testing.T) {
+	initArtifactoryTest(t)
+	symlinkTarget := filepath.Join(tests.GetTestResourcesPath(), "a", "a2.in")
+	// Path to local file with a different name from symlinkTarget
+	testFile := filepath.Join(tests.GetTestResourcesPath(), "a", "a1.in")
+	tmpDir, err := fileutils.CreateTempDir()
+	assert.NoError(t, err)
+	defer os.RemoveAll(tmpDir)
+	// Link valid symLink to local file
+	err = os.Symlink(symlinkTarget, filepath.Join(tmpDir, "symlink"))
+	assert.NoError(t, err)
+	err = fileutils.CopyFile(tmpDir, testFile)
+	// Upload symlink and local file to artifactory
+	assert.NoError(t, artifactoryCli.Exec("u", tmpDir+"/*", tests.RtRepo1+"/test-archive.zip", "--archive=zip", "--symlinks=true"))
+	os.RemoveAll(tmpDir)
```

Comment:
Let's assert the error response receieved from ``` os.RemoveAll```.
There are a few other places in the code in which the error is not asserted. Let's take this opportunity and fix them as well.